ctms/bin/acoustic.py

```diff
@@ -0,0 +1,150 @@
+    """CTMS command to sync contacts with Acoustic."""
+    return do_resync(ctx.obj["dbsession"], email_list, newsletter, waitlist)
+
+
+@cli.group(help="Manage Acoustic fields")
+@click.pass_context
+def fields(ctx):
+    pass
+
+
+@fields.command(name="list")
+@click.option(
+    "-t", "--tablename", default="main", help="Acoustic table name (default: 'main')"
+)
+@click.pass_context
+def fields_list(ctx, tablename):
+    entries = get_all_acoustic_fields(ctx.obj["dbsession"])
+    print("\n".join(sorted(f"- {e.tablename}.{e.field}" for e in entries)))
+
+
+@fields.command(name="add")
+@click.option(
+    "-t", "--tablename", default="main", help="Acoustic table name (default: 'main')"
+)
+@click.argument("field")
+@click.pass_context
+def fields_add(ctx, field, tablename):
+    row = create_acoustic_field(ctx.obj["dbsession"], tablename, field)
+    print(f"Added '{row.tablename}.{row.field}'.")
+
+
+@fields.command(name="remove")
+@click.option(
+    "-t", "--tablename", default="main", help="Acoustic table name (default: 'main')"
+)
+@click.argument("field")
+@click.pass_context
+def fields_remove(ctx, field, tablename):
+    row = delete_acoustic_field(ctx.obj["dbsession"], tablename, field)
+    if not row:
+        print(f"Unknown field '{tablename}.{field}'. Give up.")
+        return os.EX_DATAERR
+    print(f"Removed '{row.tablename}.{row.field}'.")
+    return os.EX_OK
+
+
+@cli.group(name="newsletter-mappings", help="Manage newsletter Acoustic mappings")
+@click.pass_context
+def newsletter_mappings(ctx):
+    pass
+
+
+@newsletter_mappings.command(name="list")
+@click.pass_context
+def newsletter_mappings_list(ctx):
+    entries = get_all_acoustic_newsletters_mapping(ctx.obj["dbsession"])
+    print("\n".join(sorted(f"- {e.source!r} → {e.destination!r}" for e in entries)))
+    return os.EX_OK
+
+
+@newsletter_mappings.command(
+    name="add", help='Specified as "<newsletter-name>:<acoustic-column>"'
+)
+@click.argument("mapping")
+@click.pass_context
+def newsletter_mappings_add(ctx, mapping):
+    source, destination = mapping.split(":")
+    # This will fail if mapping already exists.
+    create_acoustic_newsletters_mapping(ctx.obj["dbsession"], source, destination)
+    print(f"Added {source!r} → {destination!r}.")
+    return os.EX_OK
+
+
+@newsletter_mappings.command(name="remove")
+@click.argument("source")
+@click.pass_context
+def newsletter_mappings_remove(ctx, source):
+    row = delete_acoustic_newsletters_mapping(ctx.obj["dbsession"], source)
+    if not row:
+        print(f"Unknown mapping '{source}'. Give up.")
+        return os.EX_DATAERR
+    print(f"Removed {row.source!r} → {row.destination!r}.")
+    return os.EX_OK
+
+
+def do_resync(dbsession, email_list=None, newsletter=None, waitlist=None):
+    to_resync = []
+    if email_list:
+        for line in email_list.readlines():
+            to_resync.append(line.decode("utf-8").rstrip().lower())
```

Comment:
Do we need `line.decode("utf-8")`? I think `readlines` will automatically read the file as a list of strings.